Automattic/wp-calypso
# Plans: Refactor DisplayPrice on the JetpackProductCard component

#### Changes proposed in this Pull Request

* Move the `DisplayPrice` component out of the same file as `JetpackProductCard`, and into its own directory.
* Split the `DisplayPrice` component into a collection of components that each handle one state, with a wrapper to determine which state should be shown.
* Move all relevant styles into a new `style.scss` file inside the `DisplayPrice` component directory.
* Rename all relevant styles to adhere to ESLint coding standards (i.e., styles are prepended with `display-price` instead of `jetpack-product-card__price`).

#### Testing instructions

**This is a refactor PR,** meaning it should have no effect on existing page behavior or appearance. In Calypso Blue, Calypso Green, and Jetpack Connect, verify the price display matches production for the following cases:

* products that are **free**;
* products that are **paid**;
* products **included** in the selected site's plan;
* products **owned** by the selected site;
* products that are **deprecated (e.g., Jetpack Personal)**; and
* products with a **price range (e.g., Jetpack Search)**.
